WordPress/Classes/Plugins/Views/PluginDetailsView.swift

```diff
@@ -390,19 +414,52 @@ enum ActionButton {
     }
 }
 
+private enum PluginOperation: Hashable {
+    case install
+    case uninstall
+    case activate
+    case deactivate
+}
+
+private struct PluginOperationStatus {
+    var operation: PluginOperation
+    var result: Result<Void, Error>?
+
+    var isCompleted: Bool {
+        result != nil
+    }
+
+    var errorMessage: String? {
+        if case let .failure(error)? = result {
+            return (error as? WpApiError)?.errorMessage ?? error.localizedDescription
+        }
+        return nil
+    }
+}
+
 @MainActor
 final class WordPressPluginDetailViewModel: ObservableObject {
     let slug: PluginWpOrgDirectorySlug
     let service: PluginServiceProtocol
 
     @Published private(set) var isLoading = false
-    @Published private(set) var isUninstalling = false
-    @Published private(set) var isInstalling = false
     @Published private(set) var plugin: PluginInformation?
     @Published private(set) var installed: InstalledPlugin?
     @Published private(set) var error: String?
-    @Published private(set) var isActivating = false
-    @Published private(set) var isDeactivating = false
+
+    @Published private(set) fileprivate var operation: PluginOperationStatus?
+    var isUninstalling: Bool {
```

Comment:
With the addition of `PluginOperationStatus`, it seems there is now a good opportunity to eliminate these individual properties and also switch over the enum in the View:

```swift
if viewModel.isUninstalling {
                     inlineProgressView(title: Strings.uninstallingTitle, message: Strings.uninstallingMessage)
                 } else if viewModel.isInstalling {
```

Reply:
Good call! Addressed in https://github.com/wordpress-mobile/WordPress-iOS/pull/24092/commits/16d3194005cd3b446d2af10804b8b44b73d2d5ed.